ecommerce/views.py

```diff
@@ -124,8 +128,25 @@ def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
             order.status = Order.FULFILLED
 
         order.save_and_log(acting_user=None)
+
+        run = order.get_bootcamp_run()
+        if is_paid_in_full(user=order.user, run_key=run.run_key):
+            BootcampRunEnrollment.objects.get_or_create(
+                user=order.user,
+                bootcamp_run=run,
+            )
+
+            try:
+                application = order.applications.get()
+                application.complete()
+                application.save()
+            except BootcampApplication.DoesNotExist:
+                log.exception("Missing application for order %d. Unable to set application state.", order.id)
+            except TransitionNotAllowed:
+                log.exception("Application received full payment but state cannot transition to COMPLETE")
```

Comment:
This view is hefty as it is. Please create a separate API function for this. Something like `complete_order(order, run)`. I also feel like it would make sense to move the code that sets `order.status = Order.FULFILLED` into that function as well since that's one of the side effects we want when an order is successfully completed. If that makes sense and it's not a problem code-wise, we could change the `if` statement that runs before completing the order: `if decision == CYBERSOURCE_DECISION_ACCEPT and is_paid_in_full(user=order.user, bootcamp_run=run):`

Reply:
I made most of these changes but I left the `order.status = Order.FULFILLED` part out of `complete_order` for consistency with xPRO and other similar ecommerce logic.